### PR TITLE
Start bugfix | Add config-related changes

### PR DESCRIPTION
**Pull request by taimoorgit**
## Issues

https://github.com/OpsLevel/team-platform/issues/278

https://github.com/OpsLevel/team-platform/issues/276

## Changelog

- [x] Remove duplicate repositories from config
- [x] Re-add system to config
- [x] Correctly display quotes around version, so it is not read as a number
- [x] If user forgot their config file, gently remind them instead of `panic()`
- [x] Make a `changie` entry

## Tophatting

```
$ go run main.go config sample
version: "1.3.0"
service:
  import:
    - selector:
        apiVersion: apps/v1
        kind: Deployment
        excludes:
          - .metadata.namespace == "kube-system"
          - .metadata.annotations."opslevel.com/ignore"
      opslevel:
        aliases:
          - '"k8s:\(.metadata.name)-\(.metadata.namespace)"'
          - '"\(.metadata.namespace)-\(.metadata.name)"'
        description: .metadata.annotations."opslevel.com/description"
        framework: .metadata.annotations."opslevel.com/framework"
        language: .metadata.annotations."opslevel.com/language"
        lifecycle: .metadata.annotations."opslevel.com/lifecycle"
        name: .metadata.name
        owner: .metadata.annotations."opslevel.com/owner"
        product: .metadata.annotations."opslevel.com/product"
        properties:
          prop_object: .metadata.annotations.prop_object
        repositories:
          - '{"name": "My Cool Repo", "directory": "", "repo": .metadata.annotations.repository} | if .repo then . else empty end'
          - .metadata.annotations.repo
          - '.metadata.annotations | to_entries |  map(select(.key | startswith("opslevel.com/repo"))) | map({"name": .key | split(".")[2], "directory": .key | split(".")[3:] | join("/"), "repo": .value})'
        system: .metadata.annotations."opslevel.com/system"
        tags:
          assign:
            - '{"imported": "kubectl-opslevel"}'
            - '.metadata.annotations | to_entries |  map(select(.key | startswith("opslevel.com/tags"))) | map({(.key | split(".")[2]): .value})'
            - .metadata.labels
          create:
            - '{"environment": .spec.template.metadata.labels.environment}'
        tier: .metadata.annotations."opslevel.com/tier"
        tools:
          - '{"category": "other", "environment": "production", "displayName": "my-cool-tool", "url": .metadata.annotations."example.com/my-cool-tool"} | if .url then . else empty end'
          - '.metadata.annotations | to_entries |  map(select(.key | startswith("opslevel.com/tools"))) | map({"category": .key | split(".")[2], "displayName": .key | split(".")[3], "url": .value})'
```

```
$ go run main.go config view
version: "1.3.0"
service:
  import:
    - selector:
        apiVersion: apps/v1
        kind: Deployment
        excludes:
          - .metadata.namespace == "kube-system"
          - .metadata.annotations."opslevel.com/ignore"
      opslevel:
        aliases:
          - '"k8s:\(.metadata.name)-\(.metadata.namespace)"'
        description: ""
        framework: ""
        language: ""
        lifecycle: ""
        name: .metadata.name
        owner: .metadata.namespace
        product: ""
        properties:
          prop_string: .metadata.annotations.prop_string
        repositories: []
        system: ""
        tags:
          assign:
            - '{"imported": "kubectl-opslevel"}'
            - .metadata.labels
          create:
            - '{"environment": .spec.template.metadata.labels.environment}'
        tier: ""
        tools: []


```

### CODE REVIEW

File: src/cmd/config.go [hunk 1]
```diff
@@ -75,20 +93,22 @@ func init() {
 }
 
 func readConfig() []byte {
-	var err error
-	var res []byte
-
-	switch cfgFile {
-	case ".":
-		res, err = os.ReadFile("./opslevel-k8s.yaml")
```
Comment (taimoorgit) [inline src/cmd/config.go:83]: This situation should not be possible

1. You cannot provide `.` as a CLI argument (unless you do some weird trickery. It's not a good practice.)
2. This is never set to `.` by default - it's set to `./opslevel-k8s.yaml` [so this code is misleading.](https://github.com/OpsLevel/kubectl-opslevel/blob/c3a97ad6af097f2f82765a7f79965e4b2570e6ca/src/cmd/root.go#L45)

File: Taskfile.yml [hunk 2]
```diff
@@ -45,7 +45,6 @@ tasks:
       - gofumpt -w .
       - go mod tidy
       - golangci-lint run --timeout 5m --fix
-      - nilaway -fix -test=false -include-pkgs='github.com/opslevel/kubectl-opslevel' ./...
```
Comment (taimoorgit) [inline Taskfile.yml:48]: should not use -fix for this - too experimental. also remove nilaway on all pkgs since it makes CI super slow.

Comment (davidbloss) [reply Taskfile.yml:48]: Since we're still using `nilaway` in `task lint` and `task ci` by extension - how about just commenting this out?

Comment (taimoorgit) [reply Taskfile.yml:48]: @davidbloss I am removing this line

**Comment by taimoorgit** [comment]
@davidbloss afaict CI is failing because the latest opslevel-jq-parser has the properties change and this PR does not (neither does `main`)

File: Taskfile.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ tasks:
     cmds:
       - test -z "$(gofumpt -d -e . | tee /dev/stderr)"
       - golangci-lint run --timeout 5m
-      - nilaway -test=false ./...
+      - nilaway -test=false -include-pkgs='github.com/opslevel/kubectl-opslevel' ./...
```
Comment (davidbloss) [inline Taskfile.yml:38]: Nice one! 👏

Comment (taimoorgit) [reply Taskfile.yml]: Pushed commit to undo this line

File: .changes/unreleased/Bugfix-20240313-195638.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+kind: Bugfix
+body: Config files now display with version quoted
```
[the remark this answers is not in the document]
Comment (rocktavious) [reply .changes/unreleased/Bugfix-20240313-195638.yaml:2]: On top of this - `PrettyConfig` is tech debt - its brittle and not worth adding to our codebase if it truely isn't the problem and there is some other root problem that needs to be fixed.  If there really is some other problem with the number not being read as a string (which is what the config defines version as) then we should go after that rather then adding complexity here.  Especially the magic `VER_STRING` aspect is a huge code smell

File: src/cmd/config.go [hunk 5]
```diff
@@ -27,7 +29,7 @@ var configSchemaCmd = &cobra.Command{
 	Long:  "Print the jsonschema for configuration file",
 	Run: func(cmd *cobra.Command, args []string) {
 		schema := jsonschema.Reflect(&common.Config{})
-		jsonBytes, err := json.MarshalIndent(schema, "", "  ")
+		jsonBytes, err := json.MarshalIndent(schema, "", "    ")
```
Comment (rocktavious) [inline src/cmd/config.go:32]: why is this necessary?

File: src/common/reconciler.go [hunk 6]
```diff
@@ -412,24 +412,13 @@ func (r *ServiceReconciler) handleRepositories(service *opslevel.Service, regist
...
-			Owner:      *owner,
-			Value:      *value,
-		}
-		err = r.client.AssignPropertyHandler(input)
-		if err != nil {
-			log.Error().Err(err).Msgf("[%s] Failed assigning property with definition: '%s' and value: '%s'", service.Name, def, val)
-			continue
-		}
-		log.Info().Msgf("[%s] Successfully assigned property with definition: '%s' and value: '%s'", service.Name, def, val)
+		log.Info().Msgf("[%s] Successfully assigned property with definition: '%s' and value: '%s'", service.Name, *propertyInput.Definition.Alias, propertyInput.Value)
```
Comment (rocktavious) [inline src/common/reconciler.go:422]: could this result in a nil deference error for `propertyInput.Definition.Alias` ?

File: src/common/reconciler_test.go [hunk 7]
```diff
@@ -441,20 +439,32 @@ func Test_Reconciler_HandleTools(t *testing.T) {
...
 	}
 	registration := opslevel_jq_parser.ServiceRegistration{
 		Aliases:    []string{"a_test_service_with_properties"},
+		Name:       "A test service with properties",
 		Properties: props,
 	}
 	service := opslevel.Service{
 		ServiceId: opslevel.ServiceId{
-			Id: opslevel.ID("Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS85NzAyMg"),
+			Id: opslevel.ID("XXX"),
```
Comment (rocktavious) [inline src/common/reconciler_test.go:467]: why are we doing this?

File: src/common/reconciler_test.go [hunk 8]
```diff
@@ -477,24 +487,6 @@ func Test_Reconciler_HandleProperties(t *testing.T) {
 	// Assert
 	expLen, gotLen := len(props), len(results)
 	autopilot.Assert(t, gotLen == expLen, fmt.Sprintf("expected to get %d property assignments got %d", expLen, gotLen))
-	for _, x := range results {
```
Comment (rocktavious) [inline src/common/reconciler_test.go:480]: why is this portion of the tests being deleted?  They seem relevant still

File: src/common/reconciler.go (hunk 6 above)
Comment (taimoorgit) [reply src/common/reconciler.go:422]: @rocktavious I added a check for this in https://github.com/OpsLevel/kubectl-opslevel/pull/280

File: src/common/reconciler_test.go (hunk 7 above)
Comment (taimoorgit) [reply src/common/reconciler_test.go:469]: @rocktavious reverted this in #280.

File: src/cmd/config.go (hunk 5 above)
Comment (taimoorgit) [reply src/cmd/config.go:32]: @rocktavious I was fixing indentation in YAML and JSON files and along the way I wanted JSON to display with 4 spaces. CLI also uses 4 spaces and while it is not a standard this is the most common.
